### PR TITLE
Set correct Windows boot sector if EFI partition table used

### PR DESCRIPTION
**Pull request by canihavethisone**
Add logic to set correct Windows boot sector if EFI partition table is used or param set true

### CODE REVIEW

File: provisioning_templates/script/windows_default_script.erb [hunk 1]
```diff
@@ -101,7 +102,11 @@ for /f %%u in ('dir /s/b C:\updates\*.msu') do dism.exe /Image:C:\ /Add-Package
 
 echo Setting the boot sector
 bootsect.exe /nt60 C:
-C:\Windows\System32\bcdboot C:\Windows /l en-US
+<% if windowsEFI == 'true' or @host.ptable.name == 'Windows default GPT EFI partition table' -%>
```
Comment (lzap) [inline provisioning_templates/script/windows_default_script.erb]:
```suggestion
<% if windowsEFI == 'true' || @host.ptable.name == 'Windows default GPT EFI partition table' -%>
```

Prefer this one.

File: provisioning_templates/script/windows_default_script.erb [hunk 2]
```diff
@@ -13,6 +13,7 @@ params:
 <%
   proxy_uri = host_param('http-proxy') ? "http://#{host_param('http-proxy')}:#{host_param('http-proxy-port') || 3128}" : nil
   proxy_string = proxy_uri ? "-e http_proxy=#{proxy_uri}" : ''
+  windowsEFI = host_param('windowsEFI') ? host_param('windowsEFI') : 'false'
```
Comment (lzap) [inline provisioning_templates/script/windows_default_script.erb]: You can do this:

```suggestion
  windowsEFI = host_param('windowsEFI', 'false')
```

Actually then you don't even need this helper variable.

**Review by lzap — CHANGES_REQUESTED** [review]
Hey, thanks. Almost there.

**Review by canihavethisone — COMMENTED** [review]
Thanks lzap, love your work!

Just to confirm, is the 'name' that which is declared _in_ the template, or the name of the template file?  They differ slightly in this case.

File: provisioning_templates/script/windows_default_script.erb [hunk 3]
```diff
@@ -13,6 +13,7 @@ params:
 <%
   proxy_uri = host_param('http-proxy') ? "http://#{host_param('http-proxy')}:#{host_param('http-proxy-port') || 3128}" : nil
   proxy_string = proxy_uri ? "-e http_proxy=#{proxy_uri}" : ''
+  windowsEFI = host_param('windowsEFI', 'false')
```
Comment (lzap) [inline provisioning_templates/script/windows_default_script.erb]: Can you use `host_param_true?` or `_false?` these macros are available. They will parse "true", "yes" and "1" etc.

**Review by lzap — CHANGES_REQUESTED** [review]
Other than that, good.

File: provisioning_templates/script/windows_default_script.erb (hunk 3 above)
Comment (canihavethisone) [reply provisioning_templates/script/windows_default_script.erb]: I have updated line 16 - is that now as you suggest?  Thanks.

**Comment by tbrisker** [comment]
On behalf of the Foreman team, thank you for your contribution to the Foreman community templates repository! 
This repository is being archived and all templates are migrated to the relevant repositories. If you wish to pursue this change, please re-open the PR against the relevant repository. For more information and background, please [click here]( https://community.theforeman.org/t/heads-up-community-templates-repo-will-be-archived-tomorrow/19539).

**Pull request closed without merging**